src/gadget/monotone.rs

```diff
@@ -0,0 +1,259 @@
+use halo2::{
+    circuit::{Chip, Layouter},
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Fixed, VirtualCells},
+    poly::Rotation,
+};
+use pasta_curves::arithmetic::FieldExt;
+use std::{marker::PhantomData, u64};
+
+#[derive(Clone, Debug)]
+pub(crate) struct MonotoneConfig {
+    range_table: Column<Fixed>,
+}
```

Comment:
Include the advice column in the Config.

```suggestion
pub(crate) struct MonotoneConfig {
    range_table: Column<Fixed>,
    value: Column<Advice>,
}
```

Reply:
Fixed in second commit